### DISCUSSION TITLE
High CPU Usage in PWA Lobechat

### DISCUSSION THREAD

**Discussion by ChenglongWang** [lobehub/lobehub#4784] [General Question] [answered]
Hi,

I am currently using the Progressive Web App (PWA) version of Lobechat for my daily usage. However, I've noticed that the CPU usage for the Lobechat client is significantly higher compared to other applications, which also caused high battery usage.

Is there a way to address this issue and reduce the CPU consumption? Any recommendations or tips would be greatly appreciated.
Machine:  MacBook Pro M4

Thank you!

<img width="659" alt="Screenshot 2024-11-25 at 12 29 13" src="https://github.com/user-attachments/assets/726f2e32-05cb-4855-8abe-c7fabbe1e6fe">

**Comment by arvinxx** [comment]
I think the possible reason is you have a large chat history context. you can refer to this: https://github.com/lobehub/lobe-chat/discussions/1029 for more info.

**Reply by ChenglongWang** [reply]
Hi, @arvinxx  Thank you for your quick response. I noticed that the process is Lobechat graphics and media. Lobechat persistent to have high CPU occupancy even in the background  with no interaction or inference. The background process is `apple.webkit.GPU`. Hope this info could help.
<img width="1029" alt="Screenshot 2024-11-25 at 21 29 51" src="https://github.com/user-attachments/assets/9733dc9c-a94a-42dd-8c95-4213fb73139a">

**Comment by arvinxx** [comment, marked as answer]
we have found it's the banner causing high CPU Usage. And only Safari cause 100%+ CPU usage, if you switch to Chrome, it will only use about 50%.

<img width="808" alt="image" src="https://github.com/user-attachments/assets/bac1e691-0431-47ac-8e99-fc41ad48a914">
